File: Web/y-web/app/Services/ApiService.php
<?php

namespace App\Services;

use Illuminate\Support\Facades\Http;

class ApiService
{
    private $baseUrl;

    public function __construct()
    {
        $this->baseUrl = 'http://localhost:8080/yapi/rest';
    }

    public function login($email, $password)
    {
        $response = Http::post($this->baseUrl . '/users/login', [
            'email' => $email,
            'password' => $password,
        ]);

        return $response;
    }

    public function register($data)
    {
        $response = Http::post($this->baseUrl . '/users/register', $data);

        return $response;
    }
    public function createPost($data)
    {
        return Http::post($this->baseUrl . '/posts/create', $data);
    }
    public function getAllPosts()
    {
        return Http::get($this->baseUrl . "/posts");
    }
    public function getFollowingPosts($userId)
    {
        return Http::get($this->baseUrl . "/posts/following/$userId");
    }

    public function getUserProfile($userId)
    {
        return Http::get($this->baseUrl . "/users/$userId");
    }

    public function getUserPosts($userId)
    {
        return Http::get($this->baseUrl . "/posts/$userId");
    }
    public function getFollowers($userId)
    {
        return Http::get($this->baseUrl . "/followers/$userId/followersCount");
    }
    public function getFollowing($userId)
    {
        return Http::get($this->baseUrl . "/followers/$userId/followingCount");
    }

    public function getNotifications($userId)
    {
        return Http::get($this->baseUrl . "/notifications/$userId");
    }
    public function deletePost($postId)
    {
        return Http::delete($this->baseUrl . "/posts/delete/$postId");
    }
    public function updateProfile($userId, $data)
    {
        return Http::put($this->baseUrl . "/users/update/$userId", $data);
    }

    public function deleteAccount($userId)
    {
        return Http::delete($this->baseUrl . "/users/delete/$userId");
    }

    public function followUser($followerId, $followingId)
    {
        return Http::post($this->baseUrl . "/followers/$followerId/follow/$followingId");
    }

    public function unfollowUser($followerId, $followingId)
    {
        return Http::delete($this->baseUrl . "/followers/$followerId/unfollow/$followingId");
    }

    public function isFollowing($followerId, $followingId)
    {
        return Http::get($this->baseUrl . "/followers/isFollowing/$followerId/$followingId");
    }
}

File: apps/desktop/YDesktop/YDesktop/Services/ApiService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using YDesktop.ApiCalls;
using YDesktop.Models;
using static System.Windows.Forms.VisualStyles.VisualStyleElement.ListView;

namespace YDesktop.Services
{
    internal class ApiService
    {
        private static ApiService instance;

        public static ApiService Instance
        {
            get
            {
                if (instance == null)
                {
                    instance = new ApiService();
                }

                return instance;
            }
        }
        // USERS
        public async Task<User> Login(LoginRequest req)
        {
            return await UserController.Login(req);
        }

        public async Task<HttpResponseMessage> Register(RegisterRequest req)
        {
            return await UserController.Register(req);
        }

        public async Task<User> GetUser(int id)
        {
            return await UserController.GetUser(id);
        }

        public async Task<HttpResponseMessage> UpdateUser(int id, UserUpdate update)
        {
            return await UserController.UpdateUserAsync(id, update);
        }
        public async Task<HttpResponseMessage> DeleteUser(int id)
        {
            return await UserController.DeleteUserAsync(id);
        }

        // POSTS

        public async Task<List<Post>> GetFeed()
        {
            return await PostController.GetPostsAsync();
        }

        public async Task<List<Post>> GetUserPosts(int id)
        {
            return await PostController.GetUserPostsAsync(id);
        }

        public async Task<List<Post>> GetFollowingPosts(int id)
        {
            return await PostController.GetFollowingPostsAsync(id);
        }

        public async Task<bool> CreatePost(int userId, string content)
        {
            return await PostController.CreatePostAsync(userId, content);
        }

        public async Task<HttpResponseMessage> DeletePost(int id)
        {
            return await PostController.DeletePostAsync(id);
        }

        // FOLLOWERS

        public Task<bool> IsFollowing(int followerId, int followingId)
        {
            return FollowerController.IsFollowing(followerId, followingId);
        }

        public Task Follow(int followerId, int followingId)
        {
            return FollowerController.Follow(followerId, followingId);
        }

        public Task Unfollow(int followerId, int followingId)
        {
            return FollowerController.Unfollow(followerId, followingId);
        }

        public Task<int> GetFollowersCount(int userId)
        {
            return FollowerController.GetFollowersCount(userId);
        }

        public Task<int> GetFollowingCount(int userId)
        {
            return FollowerController.GetFollowingCount(userId);
        }

        // NOTIFICATIONS

        public async Task<List<Notification>> GetNotifications(int userId)
        {
            return await NotificationController.GetUserNotifications(userId);
        }


    }
}
